packages/opentelemetry-exporter-collector/src/transformMetrics.ts

```diff
@@ -0,0 +1,315 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CollectorMetricExporterBase } from './CollectorMetricExporterBase';
+import {
+  MetricRecord,
+  MetricKind,
+  HistogramAggregator,
+  MinMaxLastSumCountAggregator,
+  Histogram,
+  Distribution,
+} from '@opentelemetry/metrics';
+import { opentelemetryProto, CollectorExporterConfigBase } from './types';
+import * as api from '@opentelemetry/api';
+import * as core from '@opentelemetry/core';
+import { Resource } from '@opentelemetry/resources';
+import { toCollectorResource } from './transform';
+
+/**
+ * Converts labels
+ * @param labels
+ */
+export function toCollectorLabels(
+  labels: api.Labels
+): opentelemetryProto.common.v1.StringKeyValue[] {
+  return Object.entries(labels).map(([key, value]) => {
+    return { key, value };
+  });
+}
+
+/**
+ * Given a MetricDescriptor, return its type in a compatible format with the collector
+ * @param descriptor
+ */
+export function toCollectorType(
+  metric: MetricRecord
+): opentelemetryProto.metrics.v1.MetricDescriptorType {
+  if (
+    metric.descriptor.metricKind === MetricKind.COUNTER ||
+    metric.descriptor.metricKind === MetricKind.SUM_OBSERVER
+  ) {
+    if (metric.descriptor.valueType === api.ValueType.INT) {
+      return opentelemetryProto.metrics.v1.MetricDescriptorType.MONOTONIC_INT64;
+    }
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.MONOTONIC_DOUBLE;
+  }
+  if (metric.aggregator instanceof HistogramAggregator) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.HISTOGRAM;
+  }
+  if (metric.descriptor.valueType == api.ValueType.INT) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.INT64;
+  }
+  if (metric.descriptor.valueType === api.ValueType.DOUBLE) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorType.DOUBLE;
+  }
+
+  // TODO: Add Summary once implemented
+  return opentelemetryProto.metrics.v1.MetricDescriptorType.INVALID_TYPE;
+}
+
+/**
+ * Given a MetricDescriptor, return its temporality in a compatible format with the collector
+ * @param descriptor
+ */
+export function toCollectorTemporality(
+  metric: MetricRecord
+): opentelemetryProto.metrics.v1.MetricDescriptorTemporality {
+  if (
+    metric.descriptor.metricKind === MetricKind.COUNTER ||
+    metric.descriptor.metricKind === MetricKind.SUM_OBSERVER
+  ) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorTemporality.CUMULATIVE;
+  }
+  if (
+    metric.descriptor.metricKind === MetricKind.UP_DOWN_COUNTER ||
+    metric.descriptor.metricKind === MetricKind.UP_DOWN_SUM_OBSERVER
+  ) {
+    return opentelemetryProto.metrics.v1.MetricDescriptorTemporality.DELTA;
+  }
+  if (
+    metric.descriptor.metricKind === MetricKind.VALUE_OBSERVER ||
+    metric.descriptor.metricKind === MetricKind.VALUE_RECORDER
+  ) {
+    // TODO: Change once LastValueAggregator is implemented.
```

Comment:
Are you sure about this LastValueAggregator ? As recently I have removed LastValueAggregator in favour of `MinMaxLastSumCountAggregator` which is default now for value and recorder observers

Reply:
I asked within Opentelemetry-proto on what to do about temporality (link [here](https://github.com/open-telemetry/opentelemetry-proto/pull/140#issuecomment-652584080)), and I was told that `LastValueAggregator` and `ExactValueAggregator` have instantaneous temporality, whereas `MinMaxLastSumCountAggregator` is delta.

However, within my code, I just use `MinMaxLastSumCountAggregator` essentially as a `LastValueAggregator`, because I just use `last` and ignore the min, max, sum, and count. Should I just change this to use instantaneous temporality and remove the todo?

Reply:
If I understand this correctly I think the way you did it for now is fine. We just have to watch the mentioned issue as it seems like this is not a final, (had the same with LastValueAggregator).